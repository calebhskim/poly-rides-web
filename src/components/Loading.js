import React from 'react';

const Loading = () => (
  <div>
    <p>loading...</p>
  </div>
);

export default Loading;
