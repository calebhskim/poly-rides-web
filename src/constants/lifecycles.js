export default {
  AUTH_LOGGEDIN: 'AUTH_LOGGEDIN',
  AUTH_NOT_LOGGEDIN: 'AUTH_NOT_LOGGEDIN',
  AUTH_START: 'AUTH_START',
  AUTH_FAILURE: 'AUTH_FAILURE',
};
