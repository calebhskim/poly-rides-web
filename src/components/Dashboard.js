import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { Text, View } from 'react-native';

import lifecycles from '../constants/lifecycles';
import Loading from './Loading';

class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { displayName, email, loading } = this.props;

    if (loading === lifecycles.LOADING) {
      return <Loading />;
    }

    return (
      <View>
        <Text>Welcome { displayName || email }</Text>
      </View>
    );
  }
}

Dashboard.propTypes = {
  loading: PropTypes.string,
  displayName: PropTypes.string,
  email: PropTypes.string,
};

function mapStateToProps(state) {
  const { appState: { loading }, auth: { user: { displayName, email } } } = state;
  return {
    loading,
    displayName,
    email,
  };
}

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
