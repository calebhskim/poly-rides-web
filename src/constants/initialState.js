import lifecycles from './lifecycles';

export default {
  auth: {
    error: {},
    lifecycle: lifecycles.AUTH_NOT_LOGGEDIN,
    user: {
      displayName: null,
      email: null,
      emailVerified: false,
      isAnonymous: null,
      photoURL: null,
      refreshToken: null,
      uid: null,
    },
    fbToken: null,
  },
  config: {
    title: 'PolyRides',
  },
  firebase: {
    app: null,
    fbProvider: null,
  },
};
