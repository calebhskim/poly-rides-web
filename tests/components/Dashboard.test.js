import React from 'react';
import { Text } from 'react-native';

import cloneDeep from 'lodash/cloneDeep';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Dashboard from '../../src/components/Dashboard';
import initialState from '../../src/constants/initialState';
import lifecycles from '../../src/constants/lifecycles';
import Loading from '../../src/components/Loading';

const mockStore = configureMockStore();

describe('<Dashboard />', () => {
  it('renders properly', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<Dashboard store={store} />).shallow();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('shows <Loading /> when status is loading', () => {
    const modifiedState = cloneDeep(initialState);
    modifiedState.appState.status = lifecycles.LOADING;

    const store = mockStore(modifiedState);
    const wrapper = shallow(<Dashboard store={store} />).shallow();

    expect(wrapper.find(Loading)).toHaveLength(1);
    expect(wrapper.find(Text)).toHaveLength(0);
  });
  it('does not show <Loading /> when status is not loading and shows right displayName', () => {
    const modifiedState = cloneDeep(initialState);
    modifiedState.appState.status = lifecycles.DATA_LOADED;
    modifiedState.auth.user.displayName = 'bobby';

    const store = mockStore(modifiedState);
    const wrapper = shallow(<Dashboard store={store} />).shallow();

    expect(wrapper.find(Loading)).toHaveLength(0);
    expect(wrapper.find(Text)).toHaveLength(1);
    // this is pretty gross.
    // text breaks up its contents into two groups so the later one is selcted to compare
    //  to the displayname
    expect(wrapper.find(Text).children().last().text()).toEqual('bobby');
  });
});
