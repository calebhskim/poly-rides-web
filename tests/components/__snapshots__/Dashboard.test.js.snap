// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders properly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Welcome 
  </Text>
</View>
`;
